src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -150,18 +175,41 @@ public class PipelineCacheServer
             return pipelineCacheClient;
         }
 
-        private Task DownloadPipelineCacheAsync(
+        private string GetUploadPath(string preferredContentFormat, AgentTaskPluginExecutionContext context, string path, CancellationToken cancellationToken)
+        {
+            string uploadPath = path;
+            if(preferredContentFormat == ContentFormatConstants.SingleTar)
+            {
+                uploadPath = TarUtils.ArchiveFilesToTar(context, path, cancellationToken).Result;
+            }
```

Comment:
this needs to be awaited:

uploadPath = await TarUtils.ArchiveFilesToTarAsync(context, path, cancellationToken);